05.wc/wc.rb

```diff
@@ -0,0 +1,78 @@
+# ! /usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+SEPARATOR_SIZE = 1
+STDIN_FILENAME = '-'
+TOTAL = '合計'
+
+def main
+  options = parse_options(ARGV)
+  counts = []
+  ARGF.each(nil) do |input_text|
+    count = {}
+
+    count[:filename] = ARGF.filename
+    count[:line] = input_text.lines.count if options[:l]
+    count[:word] = input_text.split(/\s+/).size if options[:w]
+    count[:bytesize] = input_text.size if options[:c]
+    counts << count
+  end
+  counts << calculate_counts_total(counts) if counts.size > 1
+  max_column_widths = calculate_wc_max_column_widths(counts)
+  show_wc_rows(counts, max_column_widths)
+end
+
+def calculate_counts_total(counts)
+  counts_without_filename = counts.map { |count| count.except(:filename) }
+
+  total = counts_without_filename.inject({}) do |result, count|
+    result.merge(count) do |_key, current_val, adding_value|
+      current_val + adding_value
+    end
+  end
+  total[:filename] = TOTAL
+  total
+end
+
+def show_wc_rows(rows, max_column_widths)
+  rows.each do |row|
+    cells = {}
+    row.each do |column_name, cell|
+      cells[column_name] = if column_name == :filename
```

Comment:
```suggestion
    columns = {}
    counts.each do |name, value|
      columns[name] = if name == :filename
```

行に対応する用語は列なので、cellsよりもcolumnsの方がいいと思います。